powergenome/load_construction.py

```diff
@@ -0,0 +1,730 @@
+    "space heating and cooling",
+    "water heating",
+    "light-duty vehicles",
+    "medium-duty trucks",
+    "heavy-duty trucks",
+    "transit buses",
+]
+Nsubsector = len(running_subsector)
+
+logger = logging.getLogger(__name__)
+
+# Define function for adjusting time-difference
+def addhour(x):
+    x += 1
+    x = x.replace(8761, 1)
+    return x
+
+
+def SolveThreeUnknowns(a1, b1, c1, d1, a2, b2, c2, d2, a3, b3, c3, d3):
+    D = (
+        a1 * b2 * c3
+        + b1 * c2 * a3
+        + c1 * a2 * b3
+        - a1 * c2 * b3
+        - b1 * a2 * c3
+        - c1 * b2 * a3
+    )
+    Dx = (
+        d1 * b2 * c3
+        + b1 * c2 * d3
+        + c1 * d2 * b3
+        - d1 * c2 * b3
+        - b1 * d2 * c3
+        - c1 * b2 * d3
+    )
+    Dy = (
+        a1 * d2 * c3
+        + d1 * c2 * a3
+        + c1 * a2 * d3
+        - a1 * c2 * d3
+        - d1 * a2 * c3
+        - c1 * d2 * a3
+    )
+    Dz = (
+        a1 * b2 * d3
+        + b1 * d2 * a3
+        + d1 * a2 * b3
+        - a1 * d2 * b3
+        - b1 * a2 * d3
+        - d1 * b2 * a3
+    )
+    Sx = Dx / D
+    Sy = Dy / D
+    Sz = Dz / D
+    d = {"Sx": Sx, "Sy": Sy, "Sz": Sz}
+    return pd.DataFrame(d)
+
+
+def SolveTwoUnknowns(a1, b1, c1, a2, b2, c2):
+    D = a1 * b2 - a2 * b1
+    Dx = c1 * b2 - c2 * b1
+    Dy = a1 * c2 - a2 * c1
+    Sx = Dx / D
+    Sy = Dy / D
+    d = {"Sx": Sx, "Sy": Sy}
+    return pd.DataFrame(d)
+
+
+def CreateOutputFolder(case_folder):
+    path = case_folder / "extra_outputs"
+    if not os.path.exists(path):
+        os.makedirs(path)
+
+
+######################################
+# CreatingBaseLoad.R
+def CreateBaseLoad(
+    years: List[int],
+    regions: List[str],
+    output_folder: Path,
+    path_growthrate: Path = None,
+) -> None:
+    path_processed = path_in
+    path_result = output_folder
+
+    ## Method 3: annually
+    load_dtypes = {
+        "Year": "category",
+        "LocalHourID": "category",
+        "Sector": "category",
+        "Subsector": "category",
+        "LoadMW": np.float32,
+    }
+    model_states = pop.loc[pop["GenX.Region"].isin(regions), "State"]
+    EFS_2020_LoadProf = pd.read_parquet(path_in / "EFS_REF_load_2020.parquet")
+    total_efs_2020_load = EFS_2020_LoadProf["LoadMW"].sum()
+
+    EFS_2020_LoadProf = EFS_2020_LoadProf.loc[
+        EFS_2020_LoadProf["State"].isin(model_states), :
+    ]
+    EFS_2020_LoadProf = EFS_2020_LoadProf.astype(load_dtypes)
+    EFS_2020_LoadProf = pd.merge(EFS_2020_LoadProf, pop, on=["State"])
+    EFS_2020_LoadProf = EFS_2020_LoadProf.assign(
+        weighted=EFS_2020_LoadProf["LoadMW"] * EFS_2020_LoadProf["State Prop"]
+    )
+    EFS_2020_LoadProf = EFS_2020_LoadProf.groupby(
+        ["Year", "GenX.Region", "LocalHourID", "Sector", "Subsector"],
+        as_index=False,
+        observed=True,
+    ).agg({"weighted": "sum"})
+
+    # Read in 2019 Demand
+    Original_Load_2019 = pd.read_parquet(path_in / "ipm_load_curves_2019_EST.parquet")
```

Comment:
Need to use existing PG functions to get regional demand curves rather than importing from a saved file.